language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/json/JsonLfWriter.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.javaapi.data.codegen.json;
+
+import java.io.IOException;
+import java.io.Writer;
+
+public class JsonLfWriter {
```

Comment:
is there a reason why you're handwriting your json encoder instead of using jackson’s JsonGenerator?

Reply:
This type will be visible in the code-gen, so it's supposed to hide the implementation, similarly to the JsonLfReader. I could have had it utilise JsonGenerator, just as JsonLfReader uses JsonParser internally, but so far it's so simple that it actually hasn't needed it. The only place we're currently using Jackson in the encode path is to escape text.

https://github.com/digital-asset/daml/pull/17603/files#diff-46522965ee2c16de1ad0ae08911a2453fa182cff1afad169d98ba62b7f431b04R41